### PR TITLE
Internal annotations 

### PR DESCRIPTION
**Pull request by BarkingBad**
image
obraz
obraz

### CODE REVIEW

File: plugins/base/src/main/kotlin/signatures/JvmSignatureUtils.kt [hunk 1]
```diff
@@ -128,6 +129,12 @@ interface JvmSignatureUtils {
             text(Typography.nbsp.toString())
         }
     }
+
+    fun <T : Documentable> WithExtraProperties<T>.stylesForDeprecated(sourceSetData: SourceSetData): Set<TextStyle> =
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/signatures/JvmSignatureUtils.kt:133]: Function name is a little bit misleading IMO 🤔 
It says "stylesForDeprecated", but the inspects the source extras to see whether or not it is even deprecated.

Suggestion: `stylesIfDeprecated`

File: plugins/base/src/main/kotlin/signatures/KotlinSignatureUtils.kt [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.*
 import org.jetbrains.dokka.model.properties.WithExtraProperties
 
 object KotlinSignatureUtils : JvmSignatureUtils {
 
     private val strategy = OnlyOnce
     private val listBrackets = Pair('[', ']')
     private val classExtension = "::class"
+    val ignoredAnnotations = setOf(
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/signatures/KotlinSignatureUtils.kt:13]: Can and should be private, right? 🤔

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 3]
```diff
@@ -694,18 +697,15 @@ private class DokkaDescriptorVisitor(
...
-                    Annotations.Annotation(
-                        it,
-                        allValueArguments.map { it.key.asString() to it.value.toValue() }.filter {
-                            it.second != null
-                        }.toMap() as Map<String, AnnotationParameterValue>
-                    )
-                }
-        else
-            null
+        DRI.from(annotationClass as DeclarationDescriptor).let {
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt:700]: There is absolutely no need for a scoping function here!

```kotlin
 private fun AnnotationDescriptor.toAnnotation2(): Annotations.Annotation? = Annotations.Annotation(
        DRI.from(annotationClass as DeclarationDescriptor),
        allValueArguments.map { it.key.asString() to it.value.toValue() }.filter {
            it.second != null
        }.toMap() as Map<String, AnnotationParameterValue>,
        annotationClass!!.annotations.hasAnnotation(FqName("kotlin.annotation.MustBeDocumented"))
    )
```

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 4]
```diff
@@ -694,18 +697,15 @@ private class DokkaDescriptorVisitor(
...
+            Annotations.Annotation(
+                it,
+                allValueArguments.map { it.key.asString() to it.value.toValue() }.filter {
+                    it.second != null
+                }.toMap() as Map<String, AnnotationParameterValue>,
+                annotationClass!!.annotations.hasAnnotation(FqName("kotlin.annotation.MustBeDocumented"))
+            )
+        }
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt:708]: Suggested 

```kotlin
private fun AnnotationDescriptor.toAnnotation(): Annotations.Annotation? {
        return Annotations.Annotation(
            dri = DRI.from(annotationClass as DeclarationDescriptor),
            params = allValueArguments
                .mapValues { (_, annotationParameterValue) -> annotationParameterValue.toValue() }
                .filterValues { annotationParameterValue -> annotationParameterValue != null }
                .cast(),
            annotationClass!!.annotations.hasAnnotation(FqName("kotlin.annotation.MustBeDocumented"))
        )
    }
```

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt [hunk 5]
```diff
@@ -694,18 +697,15 @@ private class DokkaDescriptorVisitor(
     }
 
     private fun AnnotationDescriptor.toAnnotation(): Annotations.Annotation? =
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt:699]: Why returning optional here? We know, that we always return something, right?

File: plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt [hunk 6]
```diff
@@ -421,25 +421,27 @@ object DefaultPsiToDocumentableTranslator : SourceToDocumentableTranslator {
...
+            is PsiReferenceExpression -> psiReference?.let { EnumValue(text ?: "", DRI.from(it)) }
             is PsiClassObjectAccessExpression -> ClassValue(
                 text ?: "",
                 DRI.from(((type as PsiImmediateClassType).parameters.single() as PsiClassReferenceType).resolve()!!)
             )
             else -> StringValue(text ?: "")
         }
 
-        private fun PsiAnnotation.toAnnotation() = driOfReference()?.let {
+        private fun PsiAnnotation.toAnnotation() = psiReference?.let {
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt:432]: You should really avoid having multiple "it"'s in scope. This makes it unnecessarily hard to read while giving you almost 0 benefit while writing!

```kotlin
private fun PsiAnnotation.toAnnotation() = psiReference?.let { psiReference -> 
```

File: plugins/base/src/main/kotlin/signatures/JvmSignatureUtils.kt (hunk 1 above)
Comment (BarkingBad) [reply plugins/base/src/main/kotlin/signatures/JvmSignatureUtils.kt:133]: Ok, will change

File: plugins/base/src/main/kotlin/signatures/KotlinSignatureUtils.kt (hunk 2 above)
Comment (BarkingBad) [reply plugins/base/src/main/kotlin/signatures/KotlinSignatureUtils.kt:13]: Yeah

File: plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt (hunk 6 above)
Comment (BarkingBad) [reply plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt:432]: Yeah, this PR is quite old. Recently I started giving names

Comment (BarkingBad) [reply plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt:432]: Will change

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt (hunk 3 above)
Comment (BarkingBad) [reply plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt:700]: Yeah, it probably is some leftover

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt (hunk 4 above)
Comment (BarkingBad) [reply plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt:708]: Will use

File: plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt (hunk 5 above)
Comment (BarkingBad) [reply plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt:699]: Previously, this function used to return null. Fixed

**Pull request merged**